tools/pythonpkg/tests/fast/arrow/test_arrow_fetch_recordbatch.py

```diff
@@ -1,41 +1,187 @@
+        correct = duckdb_cursor.execute("select * from t").fetchall()
+        assert res == correct
+
+    # Test with Struct
+    def test_record_batch_next_batch_struct(self, duckdb_cursor):
+        duckdb_cursor = duckdb.connect()
+        duckdb_cursor_check = duckdb.connect()
+        duckdb_cursor.execute("CREATE table t as select {'x': i, 'y': i::varchar, 'z': i+1} as a from range(3000)  as tbl(i);")
+        query = duckdb_cursor.execute("SELECT a FROM t")
+        record_batch_reader = query.fetch_record_batch(1024)
+        assert record_batch_reader.schema.names == ['a']
+        chunk = record_batch_reader.read_next_batch()
+        assert(len(chunk) == 1024)
+        chunk = record_batch_reader.read_next_batch()
+        assert(len(chunk) == 1024)
+        chunk = record_batch_reader.read_next_batch()
+        assert(len(chunk) == 952)
+        with pytest.raises(StopIteration):
+            chunk = record_batch_reader.read_next_batch()
+
+        # Check if we are producing the correct thing
+        query = duckdb_cursor.execute("SELECT a FROM t")
+        record_batch_reader = query.fetch_record_batch(1024)
+
+        res = duckdb_cursor_check.execute("select * from record_batch_reader").fetchall()
+        correct = duckdb_cursor.execute("select * from t").fetchall()
+        assert res == correct
+
+    # Test with List
+    def test_record_batch_next_batch_list(self, duckdb_cursor):
+        duckdb_cursor = duckdb.connect()
+        duckdb_cursor_check = duckdb.connect()
+        duckdb_cursor.execute("CREATE table t as select [i,i+1] as a from range(3000)  as tbl(i);")
+        query = duckdb_cursor.execute("SELECT a FROM t")
+        record_batch_reader = query.fetch_record_batch(1024)
+        assert record_batch_reader.schema.names == ['a']
+        chunk = record_batch_reader.read_next_batch()
+        assert(len(chunk) == 1024)
+        chunk = record_batch_reader.read_next_batch()
+        assert(len(chunk) == 1024)
+        chunk = record_batch_reader.read_next_batch()
+        assert(len(chunk) == 952)
+        with pytest.raises(StopIteration):
+            chunk = record_batch_reader.read_next_batch()
+
+        # Check if we are producing the correct thing
+        query = duckdb_cursor.execute("SELECT a FROM t")
+        record_batch_reader = query.fetch_record_batch(1024)
+
+        res = duckdb_cursor_check.execute("select * from record_batch_reader").fetchall()
+        correct = duckdb_cursor.execute("select * from t").fetchall()
+
+        assert res == correct
+
+    # Test with Map
+    def test_record_batch_next_batch_list(self, duckdb_cursor):
+        duckdb_cursor = duckdb.connect()
+        duckdb_cursor_check = duckdb.connect()
+        duckdb_cursor.execute("CREATE table t as select map([i], [i+1]) as a from range(3000)  as tbl(i);")
+        query = duckdb_cursor.execute("SELECT a FROM t")
+        record_batch_reader = query.fetch_record_batch(1024)
```

Comment:
Can we add a test that tests different sizes in a loop? e.g. `1, 2, 4, 8, 16, 32, 33, 77, 999, 999999`?